src/util/uncaughtExceptionHandler.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+var serializeError = require('serialize-error');
+
+var logger = require('../logger').getLogger('uncaughtExceptionHandler');
+var transmission = require('../tracing/transmission');
+var agentConnection = require('../agentConnection');
+var stackTraceUtil = require('./stackTrace');
+var pidStore = require('../pidStore');
+var cls = require('../tracing/cls');
+
+var uncaughtExceptionEventName = 'uncaughtException';
+var stackTraceLength = 10;
+var config;
+
+
+exports.init = function(_config) {
+  config = _config;
+  setDefaults();
+  if (config.tracing && config.tracing.stackTraceLength != null) {
+    stackTraceLength = config.tracing.stackTraceLength;
+  }
+};
+
+
+function setDefaults() {
+  config.reportUncaughtException = config.reportUncaughtException === true;
+}
+
+
+exports.activate = function() {
+  if (config.reportUncaughtException) {
+    logger.info('Reporting uncaught exceptions is enabled.');
+    process.once(uncaughtExceptionEventName, onUncaughtException);
+  } else {
+    logger.info('Reporting uncaught exceptions is disabled.');
+  }
+};
+
+
+exports.deactivate = function() {
+  process.removeListener(uncaughtExceptionEventName, onUncaughtException);
+};
+
+
+function onUncaughtException(uncaughtError) {
+  // because of the way Error.prepareStackTrace works and how error.stack is only created once and then cached it is
+  // important to create the JSON formatted stack trace first, before anything else accesses error.stack.
+  var jsonStackTrace = stackTraceUtil.getStackTraceAsJson(stackTraceLength, uncaughtError);
+  finishCurrentSpanAndReportEvent(uncaughtError, jsonStackTrace);
+  logAndRethrow(uncaughtError);
+}
+
+
+function finishCurrentSpanAndReportEvent(uncaughtError, jsonStackTrace) {
+  var spans = finishCurrentSpan(jsonStackTrace);
+  var eventPayload = createEventForUncaughtException(uncaughtError);
+  agentConnection.reportUncaughtExceptionToAgentSync(eventPayload, spans);
+}
+
+
+function createEventForUncaughtException(uncaughtError) {
+  var eventText = JSON.stringify(serializeError(uncaughtError));
+  return {
+    title: 'A Node.js process terminated abnormally due to an uncaught exception.',
+    text: eventText,
```

Comment:
We can use markdown for event texts and should leverage it to make this data more readable.

Reply:
Good point!

Reply:
Formatted error in UI:

<img width="801" alt="bildschirmfoto 2018-09-03 um 13 35 39" src="https://user-images.githubusercontent.com/73801/44984760-4c8f4d00-af7e-11e8-8eeb-76a2d97dbd9b.png">

Reply:
👍